From 3df68e0247900b32e096eefd7465fb5e836916d1 Mon Sep 17 00:00:00 2001
From: Shikha Vyaghra <vyaghras@amazon.com>
Date: Wed, 23 Aug 2023 21:09:50 +0000
Subject: [PATCH] bottlerocket: default filesystem locations

This patch was originally written by Samuel Karp <skarp@amazon.com>
(09721aa67086fde4f3cfd05d13b7ba1c4b05d379). In this updated patch
version all the settings that can be configured using config file,
have been removed from the patch.

Signed-off-by: Sean P. Kelly <seankell@amazon.com>
Signed-off-by: Shikha Vyaghra <vyaghras@amazon.com>
---
 agent/config/config.go             | 2 +-
 agent/ecscni/plugin_linux.go       | 2 +-
 agent/ecscni/plugin_unsupported.go | 2 +-
 agent/utils/license.go             | 6 +++---
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/agent/config/config.go b/agent/config/config.go
index 34339bd..0cf7426 100644
--- a/agent/config/config.go
+++ b/agent/config/config.go
@@ -122,7 +122,7 @@ const (
 	DefaultMinSupportedCNIVersion = "0.3.0"
 
 	// pauseContainerTarball is the path to the pause container tarball
-	pauseContainerTarballPath = "/images/amazon-ecs-pause.tar"
+	pauseContainerTarballPath = "/usr/lib/amazon-ecs-agent/amazon-ecs-pause.tar"
 
 	// DefaultTaskMetadataSteadyStateRate is set as 40. This is arrived from our benchmarking
 	// results where task endpoint can handle 4000 rps effectively. Here, 100 containers
diff --git a/agent/ecscni/plugin_linux.go b/agent/ecscni/plugin_linux.go
index 0d3a940..3e76060 100644
--- a/agent/ecscni/plugin_linux.go
+++ b/agent/ecscni/plugin_linux.go
@@ -34,7 +34,7 @@ import (
 const (
 	vpcCNIPluginInterfaceType = "vlan"
 	// vpcCNIPluginPath is the path of the cni plugin's log file.
-	vpcCNIPluginPath = "/log/vpc-branch-eni.log"
+	vpcCNIPluginPath = "/var/log/ecs/vpc-branch-eni.log"
 )
 
 // newCNIGuard returns a new instance of CNI guard for the CNI client.
diff --git a/agent/ecscni/plugin_unsupported.go b/agent/ecscni/plugin_unsupported.go
index 05bc3cf..a34a823 100644
--- a/agent/ecscni/plugin_unsupported.go
+++ b/agent/ecscni/plugin_unsupported.go
@@ -26,7 +26,7 @@ import (
 
 const (
 	// vpcCNIPluginPath is the path of the cni plugin's log file.
-	vpcCNIPluginPath = "/log/vpc-branch-eni.log"
+	vpcCNIPluginPath = "/var/log/ecs/vpc-branch-eni.log"
 )
 
 // newCNIGuard returns a new instance of CNI guard for the CNI client.
diff --git a/agent/utils/license.go b/agent/utils/license.go
index 6eccff6..324307c 100644
--- a/agent/utils/license.go
+++ b/agent/utils/license.go
@@ -24,9 +24,9 @@ type LicenseProvider interface {
 type licenseProvider struct{}
 
 const (
-	licenseFile    = "LICENSE"
-	noticeFile     = "NOTICE"
-	thirdPartyFile = "THIRD-PARTY"
+	licenseFile    = "/usr/share/licenses/ecs-agent/LICENSE"
+	noticeFile     = "/usr/share/licenses/ecs-agent/NOTICE"
+	thirdPartyFile = "/usr/share/licenses/ecs-agent/THIRD_PARTY.md"
 )
 
 var readFile = ioutil.ReadFile
-- 
2.40.1

